### PR TITLE
GCE: Don't remove (shared) DNS Zone

### PR DESCRIPTION
**Pull request by justinsb**
We don't create it, we shouldn't be deleting it.

### CODE REVIEW

File: pkg/resources/gce.go [hunk 1]
```diff
@@ -735,24 +734,25 @@ func (d *clusterDiscoveryGCE) matchesClusterName(name string) bool {
...
-			Name:    zone.Name(),
-			ID:      zone.Name(),
-			Type:    "DNS Zone",
-			Deleter: d.deleteDNSZone,
-			Obj:     zone,
-		},
-	}, nil
+	// We never delete the hosted zone, because it is usually shared and we don't create it
+	return nil, nil
+	//if dns.IsGossipHostname(d.clusterName) {
```
Comment (chrislovecnm) [inline pkg/resources/gce.go]: Can we just remove the code?

Comment (justinsb) [reply pkg/resources/gce.go]: I want to keep it so that when we merge the Shared-resource PR we can reintroduce it.

Will add a comment.

**Comment by justinsb** [comment]
Added comment PTAL @chrislovecnm

**Comment by chrislovecnm** [comment]
/lgtm

Don't like commented code,  but I will live with it for now ;)

**Pull request merged**